components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java

```diff
@@ -1382,6 +1382,11 @@ public Policy[] getPolicies(int tenantId, String level) throws APIManagementExce
         return policies;
     }
 
+    @Override
+    public boolean isEnvNameExist(String name, String organization) throws APIManagementException {
+        return ChoreoApiMgtDAO.getInstance().isEnvNameExists(name, organization);
```

Comment:
In here shall we check whether the provided environment name included in the read-only environments using this method[1] as well?

[1] https://github.com/wso2/carbon-apimgt/blob/e65f6a3474235a4a9b98de16e3f9c9eed7e2600f/components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java#L7800-L7803

The reason for this is as below:
Take a scenario where a customer who uses both shared dataplane and private dataplane. Let's say shared dataplane has a read-only environment name as 'production' and customer created an environment with the same name in his private dataplane. In this scenario when we try to deploy an API in both these environments, it will only get deployed in customers' private dataplane environment only as the `getEnvironments` method[2] will prioritize customer environment name over shared dataplane read-only environment here [3].

[2] https://github.com/wso2/carbon-apimgt/blob/e65f6a3474235a4a9b98de16e3f9c9eed7e2600f/components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java#L7744-L7753

[3] https://github.com/wso2/carbon-apimgt/blob/69acb5f0a33abb727b88c378d2392d0fe89d14e9/components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/impl/ApisApiServiceImpl.java#L119-L139

Reply:
CCing @malinthaprasan for more on this